lib/money/currency.rb

```diff
@@ -14,7 +14,9 @@ class << self
       def new(currency_iso)
         raise UnknownCurrency, "Currency can't be blank" if currency_iso.nil? || currency_iso.empty?
         iso = currency_iso.to_s.downcase
-        @@loaded_currencies[iso] || @@mutex.synchronize { @@loaded_currencies[iso] = super(iso) }
+        @@loaded_currencies[iso] || @@mutex.synchronize do
+          @@loaded_currencies[iso] = iso == 'xxx' ? NullCurrency.new : super(iso)
```

Comment:
Yeah, so I was also looking at how this one line is already pretty complicated. So if we can pull out the check for a NullCurrency to before the `@@mutex.synchronize` we save having to do a synchronize on the first NullCurrency and this line becomes easier to read.

```ruby
return NullCurrency.new if iso == 'xxx'
@@loaded_currencies[iso] || @@mutex.synchronize { @@loaded_currencies[iso] = super(iso) }
```